core/trino-spi/src/main/java/io/trino/spi/block/DictionaryBlock.java

```diff
@@ -322,6 +322,16 @@ public Block copyPositions(int[] positions, int offset, int length)
     {
         checkArrayRange(positions, offset, length);
 
+        if (uniqueIds == positionCount) {
```

Comment:
With this change should we introduce  another benchmark method.  I believe copyPositions only benchmarks this short-circuit path now, and doesn't cover the generic case anymore.